src/index.js

```diff
@@ -15,6 +17,42 @@ function getHashParameters() {
   return parts;
 }
 
+function setupListener(render) {
+  hashSubs = [];
+
+  window.addEventListener('hashchange', () => {
+    render();
+    const hash = getHash();
+    const parameters = getHashParameters();
+    hashSubs.forEach(sub => {
+      setTimeout(sub, 0, {hash, parameters});
+    });
+  });
+}
+
+/**
+ * Can add a handler to be called when the
+ * hash changes.
+ *
+ * On a change event, the handler will be
+ * called with an object that contains the
+ * current `hash` and `parameters`.
```

Comment:
I don't think it really is better.  I think I was channelling my experience with 'ui-router' when I did that.  I'm happy to change it to individual arguments.

Reply:
I suppose I prefer separate parameters because there are only two and many listeners will only care about the `hash` parameter, so they can choose to only declare that parameter. If you can make that change, I'll merge your PR. Thanks for submitting this PR!